pulumi/src/index.ts

```diff
@@ -58,11 +66,34 @@ export interface ServiceConfig {
   storageSize: string
 }
 
+export interface ServiceInput {
+  coinServiceName: string
+  ports: Record<string, Port>
+  command?: Array<string>
+  args?: Array<string>
+  env?: Record<string, string>
+  dataDir?: string
+  configMapData?: Record<string, string>
+  volumeMounts?: Array<k8s.types.input.core.v1.VolumeMount>
+  readinessProbe?: k8s.types.input.core.v1.Probe
+  livenessProbe?: k8s.types.input.core.v1.Probe
+}
+
+export interface JointCoinServiceInput extends ServiceInput, ServiceConfig {}
+
 export interface Service {
```

Comment:
Let's move out any associated types to their corresponding files to encapsulate usage

Reply:
You shouldn't really need to see what a type precisely consists of to understand a piece of code. Simultaneously, that's a good test of the abstractions we're using. If we need to inspect the type to make sense of it, that means it's not a very well-named and defined type

Reply:
Yeah, this is certainly dependent on the context. We have many `types.ts` files throughout the code base for shared types. Class args are a good example of something that makes sense to have alongside the class, or function param types as you stated. Service is not used outside of the `statefulService` file/types which is why I commented on this one specifically. Subjective for sure, but I find it easier to read/reference when there is some aspect of scope related to where the type is defined/exported rather than some file that is just a dumping ground for any and every type